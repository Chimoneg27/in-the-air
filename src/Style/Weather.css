@import url('https://fonts.googleapis.com/css2?family=Inter:wght@400;600;700&family=Open+Sans&family=Roboto&display=swap');

:root {
  --text: #130e01;
  --background: #fffaeb;
  --primary: #ff8400;
  --secondary: #fff5d6;
  --accent: #cf4307;
}

#main-wrapper {
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  background-color: var(--primary);
}

.arrow {
  width: 1.25rem;
  height: 1.25rem;
  border: solid 2px white;
  display: flex;
  justify-content: center;
  align-items: center;
}

.arrow-img {
  width: 100%;
  display: flex;
  justify-content: flex-end;
  height: 1.25rem;
  margin-top: 5px;
  border-radius: 100%;
}

.cities {
  border: 2px solid var(--accent);
  display: grid;
  grid-template-columns: repeat(2, 1fr);
}

.cities a {
  text-decoration: none;
  color: white;
}

.cities a:nth-child(4n + 2),
.cities a:nth-child(4n + 3),
.cities a:nth-child(4n + 6),
.cities a:nth-child(4n + 7),
.cities a:nth-child(4n + 10) {
  background-color: var(--accent);
}

.cities a:nth-child(4n + 1),
.cities a:nth-child(4n + 4) {
  background-color: var(--primary);
}

.cities-weather {
  padding: 2rem 1rem;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.cities-weather h2 {
  font-family: 'Roboto', sans-serif;
}

.fair,
.good,
.moderate,
.poor,
.very-poor,
.quality,
.quality-index-header {
  font-family: 'Inter', sans-serif;
  text-align: center;
}
