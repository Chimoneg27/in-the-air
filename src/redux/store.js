/* eslint-disable import/no-extraneous-dependencies */
import { configureStore } from '@reduxjs/toolkit';
import weatherReducer from './weatherSlice/weatherSlice';

const store = configureStore({
  reducer: {
    weather: weatherReducer,
  },
});

export default store;
