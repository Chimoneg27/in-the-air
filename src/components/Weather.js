import React, { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { citiesWeather } from '../redux/weatherSlice/weatherSlice';
import Nav from './Nav';
import Capitals from './Capitals';
import arrow from '../assets/arrow-right.svg';
import '../Style/Weather.css';
import '../Style/Mobile.css';

const Weather = () => {
  const capitalData = useSelector((state) => state.weather.cityData);
  const isDataFetched = useSelector((state) => state.weather.isDataFetched);
  const dispatch = useDispatch();
  console.log(capitalData);
  const [filteredCities, setFilteredCities] = useState(capitalData);
  const handleFilterChange = (filterText) => {
    const filtered = capitalData.filter(
      (city) => city.city.toLowerCase().includes(filterText.toLowerCase()),
    );
    setFilteredCities(filtered);
  };

  useEffect(() => {
    if (isDataFetched === false) {
      dispatch(citiesWeather());
    }
  }, [dispatch, isDataFetched]);

  useEffect(() => {
    setFilteredCities(capitalData);
  }, [capitalData]);

  return (
    <div id="main-wrapper">
      <Nav handleFilterChange={handleFilterChange} />
      <section className="cities flex">
        {filteredCities.map((city) => (
          <Link className="flex" to={`city/${city.id}`} key={city.id}>
            <div className="arrow-img"><img className="arrow" src={arrow} alt="arrow" /></div>
            <Capitals
              key={city.id}
              city={city.city}
              aqi={city.data.main.aqi}
            />
          </Link>
        ))}
      </section>
    </div>
  );
};

export default Weather;
