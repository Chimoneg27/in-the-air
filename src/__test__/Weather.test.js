import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Weather from '../components/Weather';

const weatherRender = (component) => render(
  <Provider store={store}>
    {component}
  </Provider>,
);

describe('Weather component', () => {
  it('render the Weather component', () => {
    weatherRender(<Weather />);
    expect(Weather).toMatchSnapshot();
  });
});
