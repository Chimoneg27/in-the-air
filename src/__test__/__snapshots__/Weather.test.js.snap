// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather component render the Weather component 1`] = `[Function]`;
